src/particles/transformation/ToFixedS.H

```diff
@@ -68,30 +68,29 @@ namespace transformation
 
             // transform momenta to dynamic units (e.g., so that momenta are
             // normalized by mc):
-            px = px*m_pzd;
-            py = py*m_pzd;
-            pt = pt*m_pzd;
+            px = px * m_pzd;
+            py = py * m_pzd;
+            pz = pz * m_pzd;
 
             // compute value of particle pt = -gamma
-            amrex::ParticleReal const arg = 1.0_prt + pow(m_pzd+pt, 2) + pow(px, 2) + pow(py, 2);
+            amrex::ParticleReal const arg = 1.0_prt + pow(m_pzd + pz, 2) + pow(px, 2) + pow(py, 2);
             AMREX_ASSERT_WITH_MESSAGE(arg > 0.0_prt, "invalid pt arg (<=0)");
             amrex::ParticleReal const ptf = arg > 0.0_prt ? -sqrt(arg) : -1.0_prt;
 
             // transform position and momentum (from fixed t to fixed s)
 
-            p.pos(0) = x - px*t/(m_pzd+pt);
+            p.pos(RealAoS::x) = x - px * z / (m_pzd + pz);
             // px = px;
-            p.pos(1) = y - py*t/(m_pzd+pt);
+            p.pos(RealAoS::y) = y - py * z / (m_pzd + pz);
             // py = py;
-            p.pos(2) = ptf*t/(m_pzd+pt);  // This now represents t.
-            pt = ptf - ptdf;                    // This now represents pt.
+            p.pos(RealAoS::t) = ptf * z / (m_pzd + pz);  // This now represents t.
+            pz = ptf - ptdf;                                   // This now represents pt.
 
```

Comment:
To clarify:  The same names of the momentum components are used for both the input and output values.  Since "pz" is used here, this notation is consistent with the meaning of the input values, but not with the meaning of the output values.  Was that the intention?

Reply:
Yes, I was not sure how to fix the output values, but instead kept a comment on them and clarified in the API docs. I can introduce alias here, but would not make it clearer, I think.

Reply:
Let me try one more :)

Reply:
Updated, let me know if that looks clear :)